runtime/runtime/src/lib.rs

```diff
@@ -1857,14 +1859,22 @@ impl ApplyState {
     fn own_congestion_info(
         &self,
         protocol_version: ProtocolVersion,
-    ) -> Result<Option<ExtendedCongestionInfo>, RuntimeError> {
+        trie: &dyn TrieAccess,
+    ) -> Result<Option<CongestionInfo>, RuntimeError> {
         if ProtocolFeature::CongestionControl.enabled(protocol_version) {
-            let congestion_info = self
-                .congestion_info
-                .get(&self.shard_id)
-                .ok_or(ContextError::MissingCongestionInfo { shard_id: self.shard_id })?;
-            Ok(Some(*congestion_info))
+            if let Some(congestion_info) = self.congestion_info.get(&self.shard_id) {
+                Ok(Some(congestion_info.congestion_info))
+            } else {
```

Comment:
nit: You can invert the logic to return early, then the method will be less nested and more readable. 
```
if !ProtocolFeature::CongestionControl.enabled(protocol_version) {
  debug_assert!(self.congestion_info.is_empty());
  return Ok(None);
}
if let Some(congestion_info) = self.congestion_info.get(&self.shard_id) {
  return Ok(Some(congestion_info.congestion_info));
}
tracing::warn!(target: "runtime","starting to bootstrap congestion info, this might take a while");
...
```